### PR TITLE
add development resources for building, testing and running the proxy service and it's dependencies

### PR DESCRIPTION
**Pull request by galxy25**
# TODOs
- [x] service development skeleton
- [x] add json structured leveled logger
- [x]  fill in initial sections for readme.md
- [x]  fill in initial sections for development.md
- [x] implement v0 version of proxy service that transparently proxies requests from it to the configured kava node endpoint
- [x] add end-to-end test for the above feature
- [x] add unit tests where relevant for the above feature

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+# download service golang dependnecies source code
+RUN go mod tidy
+
+# copy over local sources used to build service
+COPY *.go ./
+
+# build service from latest sources
+# with all compilier optimizations off to support debugging
+RUN go install  -gcflags=all="-N -l"
```
Comment (galxy25) [inline Dockerfile:24]: when it's time to deploy, will create production dockerfile that complies with optimizations & doesn't install any development only dependencies

Comment (galxy25) [reply Dockerfile]: split out production and local dockerfiles

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,86 @@
+# execute these tasks when `make` with no target is invoked
+default: unit-test reset e2e-test
+
+# declare non-file based targets to speed up target
+# invocataions by letting make know it can skip checking
+# for changes in a file
+.PHONY: lint install build publish unit-test e2e-test test up down restart reset refresh logs debug-postgres debug-cache
```
Comment (nddeluca) [inline Makefile:7]: These lists get difficult to maintain -- usually opt for defining the target as phony near the definition, though has no change in functionality.

Ex: make
```
.PHONY: lint
lint:
    ....

.PHONY: install
install:
    ...
```

File: LocalDockerfile [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# Specify base image for building service binary
+FROM golang:1.20
```
Comment (nddeluca) [inline LocalDockerfile:2]: local.Dockerfile, production.Dockerfile, etc would be the preferred naming convention.  The `.Dockefile` extension is understood by github and most editors for syntax highlighting

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+services:
+  # run postgres for proxy service to store observability metrics
+  postgres:
+    image: postgres:15
+    env_file: .env
+    ports:
+      - "${POSTGRES_HOST_PORT}:${POSTGRES_CONTAINER_PORT}"
+  # run redis for proxy service to cache responses
+  redis:
+    image: 'bitnami/redis:latest'
```
Comment (nddeluca) [inline docker-compose.yml:11]: Why not the official redis image?

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    volumes:
+      - ./docker/kava:/docker/kava
+    ports:
+      - "${KAVA_HOST_COSMOS_RPC_PORT}:${KAVA_CONTAINER_COSMOS_RPC_PORT}"
+      - "${KAVA_HOST_EVM_RPC_PORT}:${KAVA_CONTAINER_EVM_RPC_PORT}"
+  # run proxy service to observe, route, and scale requests to kava api endpoints
+  proxy:
+    build:
+      dockerfile: LocalDockerfile
+    env_file: .env
```
Comment (nddeluca) [inline docker-compose.yml:29]: I believe `.env` is automatically loaded by docker-compose

Comment (galxy25) [reply docker-compose.yml:29]: correct, but I'm an adherent to the zen of python and go proverbs of "explicit is better than implicit", that way if some none docker compose wizard has to work with this file they have one less thing to be confused about (hopefully)

File: docker-compose.yml (hunk 4 above)
Comment (galxy25) [reply docker-compose.yml:11]: I've always used bitnami images wherever possible, they do a better job of patching and updating their base images as they use them in production whereas the offical redis image is more caveat emptor

File: LocalDockerfile (hunk 3 above)
Comment (galxy25) [reply LocalDockerfile:2]: good call, changing

File: Makefile (hunk 2 above)
Comment (galxy25) [reply Makefile:7]: today I learned something new about `make`

**Pull request merged**